### ISSUE TITLE
[Bug]: v9 DatePicker compat popup is not accessible in latest Chromium

### ISSUE THREAD

**Issue by smhigley** [microsoft/fluentui#31818]
### Library

React Components / v9 (@fluentui/react-components)

### System Info

```shell
N/A
```

### Are you reporting Accessibility issue?

yes

### Reproduction

repros on the docs site

## Actual Behavior
The DatePicker popup is not in the accessibility tree (does not repro on v8, related to v9 being modal). This is caused by a leftover `aria-owns` attribute that moves the popup from the end of the DOM to within the hidden tree. This was necessary when the datepicker was non-modal, but is breaking access now that it is modal.

The practical impact is that all screen readers and voice control are unable to access the dialog on the latest version of Chromium and Safari.

## Expected Behavior
The DatePicker is accessible 😅

### Requested priority

High

### Are you willing to submit a PR to fix?

yes

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.
